test/test_jit.py

```diff
@@ -3947,6 +3947,53 @@ def test_append():
             return a == [0, 1, 2, 3]
         self.checkScript(test_append, ())
 
+    def test_comprehensions_basic(self):
+
+        @torch.jit.script
```

Comment:
use checkScript here to verify that the both frontends (AST and string) return equivalent results to python